src/Microsoft.AspNetCore.Rewrite/ConditionAbstraction/ModRewriteExpressionCreator.cs

```diff
@@ -0,0 +1,100 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+using System;
+using System.Text.RegularExpressions;
+using Microsoft.AspNetCore.Rewrite.Flags;
+using Microsoft.AspNetCore.Rewrite.Operands;
+using Microsoft.AspNetCore.Rewrite.RuleAbstraction;
+
+namespace Microsoft.AspNetCore.Rewrite.ConditionAbstraction
+{
+    public class ModRewriteExpressionCreator
+    {
+        private static readonly TimeSpan RegexTimeout = TimeSpan.FromMilliseconds(1);
+        public static ConditionExpression CreateConditionExpression(ParsedConditionExpression pce, ConditionFlags flags)
+        {
+            var condExp = new ConditionExpression();
+            condExp.Invert = pce.Invert;
+            if (pce.Type == ConditionType.Regex)
+            {
+                // todo make nullable
+                if (flags != null && flags.HasFlag(ConditionFlagType.NoCase)) {
+                    condExp.Operand = new RegexOperand { Regex = new Regex(pce.Operand, RegexOptions.IgnoreCase, RegexTimeout) };
+                }
+                else
+                {
+                    condExp.Operand = new RegexOperand { Regex = new Regex(pce.Operand, RegexOptions.None, RegexTimeout) };
+                }
+            }
+            else if (pce.Type == ConditionType.IntComp)
+            {
+                switch(pce.Operation)
+                {
+                    case OperationType.Equal:
+                        condExp.Operand = new IntegerOperand { Operation = IntegerOperationType.Equal, Value = Int32.Parse(pce.Operand) };
```

Comment:
If this fails it will throw. Is the stack trace and message such that a user can figure out what went wrong?